modules/ossm-installing-external-control-plane.adoc

```diff
@@ -0,0 +1,457 @@
+----
+
+. Set up the environment variable that contains the `oc` command context for the data plane cluster by running the following command:
++
+[source,terminal]
+----
+$ export CTX_DATA_PLANE_CLUSTER=<context_name_of_the_data_plane_cluster>
+----
+
+. Set up the ingress gateway for the control plane:
+
+.. Create a project called `istio-system` by running the following command:
++
+[source,terminal]
+----
+$ oc get project istio-system --context "${CTX_CONTROL_PLANE_CLUSTER}" || oc new-project istio-system --context "${CTX_CONTROL_PLANE_CLUSTER}"
+----
+
+.. Create an `Istio` resource on the control plane cluster to manage the ingress gateway by running the following command:
++
+[source,terminal]
+----
+$ cat <<EOF | oc --context "${CTX_CONTROL_PLANE_CLUSTER}" apply -f -
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: v${ISTIO_VERSION}
+  namespace: istio-system
+  global:
+    network: network1
+EOF
+----
+
+.. Create the ingress gateway for the control plane by running the following command:
++
+[source,terminal]
+----
+$ oc --context "${CTX_CONTROL_PLANE_CLUSTER}" apply -f https://raw.githubusercontent.com/istio-ecosystem/sail-operator/main/docs/multicluster/controlplane-gateway.yaml
+----
+
+.. Wait for the ingress gateway to be assigned an IP address by running the following command:
++
+[source,terminal]
+----
+$ oc --context "${CTX_CONTROL_PLANE_CLUSTER}" wait --for='jsonpath={.status.loadBalancer.ingress[].ip}' --timeout=300s svc istio-ingressgateway -n istio-system
+----
+
+.. Store the IP address of the ingress gateway in an environment variable by running the following command:
++
+[source,terminal]
+----
+$ export EXTERNAL_ISTIOD_ADDR=$(oc -n istio-system --context="${CTX_CONTROL_PLANE_CLUSTER}" get svc istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+----
+
+
+. Install {istio} on the data plane cluster:
+
+.. Create a project called `external-istiod` on the data plane cluster by running the following command:
++
+[source,terminal]
+----
+$ oc get project external-istiod --context "${CTX_DATA_PLANE_CLUSTER}" || oc new-project external-istiod --context "${CTX_DATA_PLANE_CLUSTER}"
+----
+
+.. Create an `Istio` resource on the data plane cluster by running the following command:
++
+[source,terminal]
+----
+$ cat <<EOF | oc --context "${CTX_DATA_PLANE_CLUSTER}" apply -f -
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: external-istiod
+spec:
+  version: v${ISTIO_VERSION}
+  namespace: external-istiod
+  profile: remote
+  values:
+    defaultRevision: external-istiod
+    global:
+      remotePilotAddress: ${EXTERNAL_ISTIOD_ADDR}
+      configCluster: true <1>
+    pilot:
+      configMap: true
+      istiodRemote:
+        injectionPath: /inject/cluster/cluster2/net/network1
+EOF
+----
+<1> This setting identifies the data plane cluster as the source of the mesh configuration.
+
```

Comment:
Create a project called `istio-cni` on the data plane cluster by running the following command:


oc get project istio-cni --context "${CTX_DATA_PLANE_CLUSTER}" || oc new-project istio-cni --context "${CTX_DATA_PLANE_CLUSTER}"

Reply:
Praneeth and I met to discuss changes.